File: backend/src/main/java/at/ltb/apprenticedeliverysystem/core/user/api/UserService.java
package at.ltb.apprenticedeliverysystem.core.user.api;

import at.ltb.apprenticedeliverysystem.core.user.dto.PortalUserDTO;
import at.ltb.apprenticedeliverysystem.core.user.dto.UpdateUserDTO;
import at.ltb.apprenticedeliverysystem.core.user.dto.UserDetailDTO;

public interface UserService {

    PortalUserDTO loadPortalUser();

    UserDetailDTO loadUser();

    UserDetailDTO updateUser(UpdateUserDTO request);
}

File: frontend/src/app/shared/service/user.service.ts
import { Injectable } from '@angular/core';
import {Observable} from "rxjs";
import {PortalUserDTO} from "../model/portalUserDTO";
import {HttpClient} from "@angular/common/http";
import {UserDetailDTO} from "../model/userDetailDTO";
import {UpdateUserDTO} from "../model/updateUserDTO";

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private _baseUrl = 'https://localhost:9002/api/user/';

  constructor(private httpClient: HttpClient) { }

  public loadLoggedInUser(): Observable<PortalUserDTO> {
    return this.httpClient.get(this._baseUrl + 'current-user');
  }

  public loadUser(): Observable<UserDetailDTO> {
    return this.httpClient.get(this._baseUrl);
  }

  public updateUser(request: UpdateUserDTO): Observable<UserDetailDTO> {
    return this.httpClient.put(this._baseUrl, request);
  }
}
